unison-syntax/src/Unison/Parser/Ann.hs

```diff
@@ -20,9 +20,28 @@ instance Monoid Ann where
   mempty = External
 
 instance Semigroup Ann where
-  Ann s1 _ <> Ann _ e2 = Ann s1 e2
+  Ann s1 e1 <> Ann s2 e2 = Ann (min s1 s2) (max e1 e2)
```

Comment:
When combining Ann's we want the full span of those Anns.

The old monoid was giving invalid Anns for things like Type arrows where the start was after the end, etc.